### PR TITLE
Add docs and PEP8 clean-up to `seaice` modules

### PR DESCRIPTION
**Pull request by xylar**
This PR adds the `seaice` modules to the documentation, adds docstrings to each "public" function, makes some functions explicitly "private" by adding a `_` in front, and reformats the code using `autopep8` and some hand editing to conform to the conventions of other modules in the package.

### CODE REVIEW

File: conda_package/docs/seaice/partition.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+The :py:mod:`mpas_tools.seaice.partition` module contains a function for
+creating graph partition files for MPAS-Seaice that are better balanced than
+those created from Metis tools directly.
+
+.. _seaice_partitions_:
+
+
+Graph partition tools
+---------------------
+`
```
Comment (akturner) [inline conda_package/docs/seaice/partition.rst:15]: Is this tilde needed?

File: conda_package/docs/seaice/partition.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+The tools ``prepare_seaice_partitions`` and ``create_seaice_partitions`` are
+used to create MPAS-Seaice graph partitions that are better balanced so that
+each processor "owns" cells both polar and equatorial regions.
```
Comment (akturner) [inline conda_package/docs/seaice/partition.rst:18]: "cells both" -> "cells from both" or equivalent

File: conda_package/docs/seaice/partition.rst [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                            Output direct
+
+The input mesh file contains the MPAS-Seaice mesh fields for the mesh used to
+create the "presence" file.  The "presence" file itself contains an
+``icePresence`` field that indicates where sea ice might be present. We
+typically use a
+`60-km mesh file <https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-seaice/partition/seaice_QU60km_polar.nc>`_
+and the corresponding
+`presence file <https://web.lcrc.anl.gov/public/e3sm/mpas_standalonedata/mpas-seaice/partition/icePresent_QU60km_polar.nc>`_.
+The presence file will be regridded to the given output MPAS-Seaice mesh.
```
Comment (akturner) [inline conda_package/docs/seaice/partition.rst:46]: Suggest adding: "Here the ice presence was determined as any cell in the mesh to have had ice present at any time during a 50 year MPAS-Seaice standalone simulation with the above 60-km mesh file."

**Review by akturner — APPROVED** [review]
Looks fabulous! I added a few minor comments to the docs.

File: conda_package/docs/seaice/partition.rst (hunk 1 above)
Comment (xylar) [reply conda_package/docs/seaice/partition.rst:15]:
```suggestion

```

Comment (xylar) [reply conda_package/docs/seaice/partition.rst:15]: Oops, definitely not! Thanks.

File: conda_package/docs/seaice/partition.rst (hunk 2 above)
Comment (xylar) [reply conda_package/docs/seaice/partition.rst:18]:
```suggestion
each processor "owns" cells from both polar and equatorial regions.
```

File: conda_package/docs/seaice/partition.rst (hunk 3 above)
Comment (xylar) [reply conda_package/docs/seaice/partition.rst:46]:
```suggestion
The presence file will be regridded to the given output MPAS-Seaice mesh. Here the ice 
presence was determined as any cell in the mesh to have had ice present at any time 
during a 50 year MPAS-Seaice standalone simulation with the above 60-km mesh file.
```

**Comment by xylar** [comment]
@akturner, thanks for the suggested edits.  I have made them all.  I have also rebased in case you want to have a look here.

I will merge once CI is done.

**Pull request merged**